domain/consensus/finality_test.go

```diff
@@ -167,3 +172,248 @@ func TestFinality(t *testing.T) {
 		}
 	})
 }
+
+func TestBoundedMergeDepth(t *testing.T) {
+	testutils.ForAllNets(t, true, func(t *testing.T, params *dagconfig.Params) {
+		// Set finalityInterval to 50 blocks, so that test runs quickly
+		params.FinalityDuration = 50 * params.TargetTimePerBlock
+		finalityInterval := int(params.FinalityDepth())
+
+		if int(params.K) >= finalityInterval {
+			t.Fatal("K must be smaller than finality duration for this test to run")
+		}
+
+		checkViolatingMergeDepth := func(consensus testapi.TestConsensus, parents []*externalapi.DomainHash) (*externalapi.DomainBlock, bool) {
+			block, err := consensus.BuildBlockWithParents(parents, nil, nil)
+			if err != nil {
+				t.Fatalf("expected err: %v, found err: %v", ruleerrors.ErrViolatingBoundedMergeDepth, err)
+				select {} // fo some reason go doesn't recognize that t.Fatalf never returns
+			}
+
+			err = consensus.ValidateAndInsertBlock(block)
+			if err == nil {
+				return block, false
+			} else if errors.Is(err, ruleerrors.ErrViolatingBoundedMergeDepth) {
+				return block, true
+			} else {
+				t.Fatalf("expected err: %v, found err: %v", ruleerrors.ErrViolatingBoundedMergeDepth, err)
+				select {} // fo some reason go doesn't recognize that t.Fatalf never returns
```

Comment:
Since it's not relevant to the test, simply return here to make it a little less complex for future readers.